// weird-dream-ui@0.1.2/lib/src/option/src/option.vue.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const vue = require("vue");
const type = require("./type.js");
require("../../../theme-chalk/src/option.css");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "option",
  props: type.optionProps,
  emits: ["value"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emits = __emit;
    const optionClass = vue.computed(() => {
      return [
        `wd-option-${props.color}`,
        `wd-option-${props.size}`
      ];
    });
    const optionValueClass = vue.computed(() => {
      return value.value === null ? "" : value.value ? [`wd-option-${props.size}-on`, `wd-option-${props.color}-on`] : [`wd-option-${props.size}-off`, `wd-option-${props.color}-off`];
    });
    const value = vue.ref(null);
    function optionChange() {
      if (value.value === null) {
        value.value = true;
      } else {
        value.value = !value.value;
      }
      emits("value", value.value);
    }
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        class: vue.normalizeClass(["wd-option", [optionClass.value, optionValueClass.value]]),
        onClick: optionChange
      }, _cache[0] || (_cache[0] = [
        vue.createElementVNode("div", { class: "wd-option-stick" }, null, -1)
      ]), 2);
    };
  }
});
exports.default = _sfc_main;
