// weird-dream-ui/es/src/slidebar/src/slidebar.vue.mjs
import { defineComponent, computed, openBlock, createElementBlock, normalizeClass } from "vue";
import { slidebarProps } from "./type.mjs";
import "../../../theme-chalk/src/slidebar.css";
const _hoisted_1 = ["min", "max", "step", "value"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "slidebar",
  props: slidebarProps,
  emits: ["value"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const slidebarClass = computed(() => {
      return [
        `wd-slidebar-${props.fill}`,
        `wd-slidebar-${props.color}-${props.fill}`,
        `wd-slidebar-${props.size}-${props.type}`
      ];
    });
    function getValue(e) {
      const element = e.target;
      const valuePersent = ((parseFloat(element.value) - parseFloat(props.min)) / (parseFloat(props.max) - parseFloat(props.min)) * 100).toFixed(2) + "%";
      element.style.setProperty("--currentValue", valuePersent);
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("input", {
        class: normalizeClass(["wd-slidebar", slidebarClass.value]),
        onInput: _cache[0] || (_cache[0] = ($event) => getValue($event)),
        onChange: _cache[1] || (_cache[1] = ($event) => getValue($event)),
        type: "range",
        min: parseFloat(props.min),
        max: parseFloat(props.max),
        step: parseFloat(props.step),
        value: parseFloat(props.value)
      }, null, 42, _hoisted_1);
    };
  }
});
export {
  _sfc_main as default
};
