// weird-dream-ui/lib/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const index$1 = require("./src/button/index.js");
const index$2 = require("./src/button-group/index.js");
const index$3 = require("./src/carousel/index.js");
const index$4 = require("./src/carousel-img/index.js");
const index$7 = require("./src/slidebar/index.js");
const index$5 = require("./src/joystick/index.js");
const index$6 = require("./src/option/index.js");
const install = function(app) {
  app.component("wdButton", index$1.wdButton);
  app.component("wdButtonGroup", index$2.wdButtonGroup);
  app.component("wdCarousel", index$3.wdCarousel);
  app.component("wdCarouselImg", index$4.wdCarouselImg);
  app.component("wdJoystick", index$5.wdJoystick);
  app.component("wdOption", index$6.wdOption);
  app.component("wdSlidebar", index$7.wdSlidebar);
};
const index = {
  install
};
exports.wdButton = index$1.wdButton;
exports.wdButtonGroup = index$2.wdButtonGroup;
exports.wdCarousel = index$3.wdCarousel;
exports.wdCarouselImg = index$4.wdCarouselImg;
exports.wdSlidebar = index$7.wdSlidebar;
exports.wdJoystick = index$5.wdJoystick;
exports.wdOption = index$6.wdOption;
exports.default = index;
